src/test/itimer/test_itimer.rs

```diff
@@ -0,0 +1,330 @@
+use std::{
+    mem::MaybeUninit,
+    ops::Sub,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use nix::sys::{
+    signal::{SaFlags, SigAction, SigHandler, SigSet, Signal},
+    time::{TimeVal, TimeValLike},
+};
+use test_utils::{ensure_ord, set, ShadowTest, TestEnvironment};
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+struct ITimer {
+    pub interval: TimeVal,
+    pub value: TimeVal,
+}
+
+impl From<libc::itimerval> for ITimer {
+    fn from(val: libc::itimerval) -> Self {
+        Self {
+            interval: TimeVal::from(val.it_interval),
+            value: TimeVal::from(val.it_value),
+        }
+    }
+}
+
+// Neither `libc` nor `nix` wrap `getitimer`.
+fn getitimer(which: i32) -> nix::Result<ITimer> {
+    let mut old_value: libc::itimerval = unsafe { MaybeUninit::zeroed().assume_init() };
+    if unsafe { libc::syscall(libc::SYS_getitimer, which, &mut old_value as *mut _) } == -1 {
+        return Err(nix::errno::Errno::last());
+    }
+    Ok(old_value.into())
+}
+
+// Neither `libc` nor `nix` wrap `setitimer`.
+fn setitimer(which: i32, new_value: &libc::itimerval) -> nix::Result<ITimer> {
+    let mut old_value: libc::itimerval = unsafe { MaybeUninit::zeroed().assume_init() };
+    if unsafe {
+        libc::syscall(
+            libc::SYS_setitimer,
+            which,
+            new_value as *const _,
+            &mut old_value as *mut _,
```

Comment:
nit: I think newer versions of rust don't require the `*const _` and `*mut _`, so removing these might make it fit on one line.

Reply:
Oh nice, done